File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        vector<int> vec1(26,0),vec2(26,0),v1(26,0),v2(26,0);
        
        for(auto a:word1)vec1[a-'a']++,v1[a-'a']=1;
        for(auto a:word2)vec2[a-'a']++,v2[a-'a']=1;
        sort(vec1.begin(),vec1.end());
        sort(vec2.begin(),vec2.end());
        return vec1==vec2&&v1==v2;
    }
};

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.java
class Solution {
  public boolean closeStrings(String word1, String word2) {
    var m1 = new int[26];
    var m2 = new int[26];
    for (var c : word1.toCharArray()) m1[c - 'a']++;
    for (var c : word2.toCharArray()) m2[c - 'a']++;
    for (var i=0; i<26; i++) {
      if (m1[i] == 0 && m2[i] != 0) return false;
      if (m1[i] != 0 && m2[i] == 0) return false;
    }
    Arrays.sort(m1);
    Arrays.sort(m2);
    return Arrays.equals(m1, m2);
  }
}